libs/libc/tls/tls_getset.c

```diff
@@ -56,16 +53,14 @@
 
 tls_ndxset_t tls_get_set(void)
 {
-  FAR struct tcb_s *rtcb = this_task();
-  FAR struct task_group_s *group = rtcb->group;
-  irqstate_t flags;
+  FAR struct task_info_s *tinfo = task_get_info();
   tls_ndxset_t tlsset;
 
-  DEBUGASSERT(group != NULL);
+  DEBUGASSERT(tinfo != NULL);
 
-  flags = spin_lock_irqsave(NULL);
-  tlsset = group->tg_tlsset;
-  spin_unlock_irqrestore(NULL, flags);
+  sem_wait(&tinfo->ta_tlssem);
```

Comment:
Unlike some of the other comments, since modification of the tlsset is not atomic, the semphore protection is probably a good idea in this case.

Reply:
Should _SEM_WAIT be used?  It should be if this could ever be called from within the OS.  If not, then it should not be built if __KERNEL__ is defined.